apps/abstraxion-dashboard/components/ModalViews/RemoveAuthenticator/RemoveAuthenticatorForm.tsx

```diff
@@ -0,0 +1,215 @@
+    AllSmartWalletQuery,
+    {
+      variables: {
+        authenticator: loginAuthenticator,
+      },
+      fetchPolicy: "network-only",
+      notifyOnNetworkStatusChange: true,
+    },
+  );
+
+  // Stop polling upon new data and update context
+  useEffect(() => {
+    if (previousData && data !== previousData) {
+      stopPolling();
+      setIsLoading(false);
+      setAbstractAccount(undefined); // set account to undefined to throw users back to account select screen
+    }
+  }, [data, previousData]);
+
+  const handleAuthenticatorLabels = (type: authenticatorTypes) => {
+    switch (type) {
+      case "SECP256K1":
+        return "OKX WALLET";
+      case "ETHWALLET":
+        return "EVM WALLET";
+      case "JWT":
+        return "EMAIL";
+      default:
+        return "";
+    }
+  };
+
+  const handleAuthenticatorLogos = (type: authenticatorTypes) => {
+    switch (type) {
+      case "SECP256K1":
+        return (
+          <img
+            className="ui-invert"
+            src="https://www.okx.com/cdn/assets/imgs/239/4A66953783FC5452.png"
+            height={24}
+            width={24}
+            alt="OKX Logo"
+          />
+        );
+      case "ETHWALLET":
+        return <EthereumLogo />;
+      case "JWT":
+        return <EmailIcon />;
+      default:
+        return <AccountWalletLogo />;
+    }
+  };
+
+  const renderAuthenticator = () => {
+    if (!authenticator) {
+      return (
+        <div className="ui-flex ui-items-center ui-px-4 ui-mb-3 ui-h-16 ui-bg-black ui-rounded-lg">
+          No authenticator found.
+        </div>
+      );
+    }
+    return (
+      <div
+        key={authenticator.id}
+        className="ui-flex ui-items-center ui-px-4 ui-mb-3 ui-h-16 ui-bg-black ui-rounded-lg"
+      >
+        <div className="ui-flex ui-w-10 ui-h-10 ui-bg-white/20 ui-items-center ui-justify-center ui-rounded-full">
+          {handleAuthenticatorLogos(
+            authenticator.type.toUpperCase() as authenticatorTypes,
+          )}
+        </div>
+        <div className="ui-ml-4 ui-flex ui-flex-1 ui-items-center ui-justify-between">
+          <p className="ui-text-white ui-text-base ui-font-normal ui-font-akkuratLL ui-leading-normal">
+            {handleAuthenticatorLabels(
+              authenticator.type.toUpperCase() as authenticatorTypes,
+            )}
+          </p>
+        </div>
+      </div>
+    );
+  };
+
+  function postAddFunction() {
+    setIsLoading(true);
+    startPolling(3000);
+  }
+
+  async function removeAuthenticator() {
+    try {
+      setIsLoading(true);
+
+      if (!authenticator) {
+        throw new Error("No authenticator found.");
+      }
+
+      if (!client) {
+        throw new Error("No client found.");
+      }
+
+      if (!abstractAccount) {
+        throw new Error("No account found.");
+      }
+
+      if (abstractAccount.authenticators.nodes.length <= 1) {
+        throw new Error("We cannot allow this operation.");
```

Comment:
If we're going to throw this error can we flesh out out the message to be more clear why this action is forbidden?